tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -171,33 +194,38 @@ private static void AddSamplingPriorityTags(Span span, int samplingPriority)
             }
         }
 
-        private void PropagateMetadata(ArraySegment<Span> spans)
+        private static void AddSamplingPriorityTags(ArraySegment<Span> spans, int samplingPriority)
         {
-            // Needs to be done for chunks as well, any span can contain the tags.
-            DecorateWithAASMetadata(spans.Array[0]);
-
             // The agent looks for the sampling priority on the first span that has no parent
             // Finding those spans is not trivial, so instead we apply the priority to every span
 
-            var samplingPriority = _samplingPriority;
-
-            if (samplingPriority == null)
+            if (spans.Array != null)
```

Comment:
I don't think we need to worry about the array being null

Reply:
No, but the compiler warnings!

Reply:
I'll revert this change and add silence the warning with `!` instead.